Formula/flow-cli.rb

```diff
@@ -0,0 +1,19 @@
+class FlowCli < Formula
+  desc "Command-line interface that provides utilities for building Flow applications"
+  homepage "https://onflow.org"
+  url "https://github.com/onflow/flow-cli/archive/v0.12.1.tar.gz"
+  sha256 "ad231ca19296ca65522eb71115900c9bd6918335caafef5913689acc1ec7effe"
+  license "Apache-2.0"
+  head "https://github.com/onflow/flow-cli.git"
+
+  depends_on "go@1.14" => :build
+
+  def install
+    system "make", "cmd/flow/flow", "VERSION=v0.12.1"
+    bin.install "cmd/flow/flow"
+  end
+
+  test do
+    system "#{bin}/flow", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Ah, I overlooked that. I've changed the test to running a hello world program in 3574403